core/Piwik.php

```diff
@@ -889,4 +889,22 @@ public static function translate($translationId, $args = array(), $language = nu
 
         return $translator->translate($translationId, $args, $language);
     }
+
+    public static function getPeriod($paramName = 'period', $default = false)
+    {
+        $period = Common::getRequestVar($paramName, $default);
```

Comment:
```suggestion
        $period = Common::getRequestVar('period', $default, 'string');
```

Using that one, will automatically throw an exception if period is provided as an array and convert it to string otherwise. If at some point we don't want to throw and exception if the period is missing we can provide an empty string as default.